prowler/providers/azure/services/vm/vm_service.py

```diff
@@ -32,29 +62,19 @@ def _get_virtual_machines(self):
                                 storage_profile=(
                                     StorageProfile(
                                         os_disk=OSDisk(
-                                            name=vm.storage_profile.os_disk.name,
-                                            managed_disk=vm.storage_profile.os_disk.managed_disk,
+                                            name=getattr(os_disk, "name", None),
+                                            managed_disk=getattr(
+                                                os_disk, "managed_disk", None
+                                            ),
                                         ),
-                                        data_disks=[
-                                            DataDisk(
-                                                lun=data_disk.lun,
-                                                name=data_disk.name,
-                                                managed_disk=data_disk.managed_disk,
-                                            )
-                                            for data_disk in getattr(
-                                                vm.storage_profile, "data_disks", []
-                                            )
-                                        ],
+                                        data_disks=data_disks,
                                     )
-                                    if getattr(vm, "storage_profile", None)
+                                    if storage_profile
                                     else None
```

Comment:
```suggestion
                                    if getattr(vm, "storage_profile", None)
```
That is what I was talking about 😄